src/widget/wlibrary.cpp

```diff
@@ -142,3 +150,10 @@ bool WLibrary::event(QEvent* pEvent) {
     }
     return QStackedWidget::event(pEvent);
 }
+
+void WLibrary::keyPressEvent(QKeyEvent* pEvent) {
+    if (pEvent->key() == Qt::Key_Left && pEvent->modifiers() & Qt::ControlModifier) {
+        emit setLibraryFocus(FocusWidget::Sidebar);
```

Comment:
This is not how I expected the feature to work, it conflicts/overlaps with `WLibraryTableView::moveCursor`:
Ctrl+Left moves the cursor until it hits the left-most cell, **then** focus is moved to the sidebar, which kind of defeats the purpose of having alternative to Tab / Shift+Tab.

Tbh I don't see the benefit of `WLibraryTableView::moveCursor` handling Ctrl+Left/Right. If it's about (implicit) horzontal scrolling we could map that to Alt+Left/Right (didn't investigate what the standard behaviour is), and ignore Ctrl+Left/Right.

What do you think?

Reply:
I just rebased the code, and I don't see what you mean. Without this key handling in wlibrary.cpp ctrl-left/right does nothing.

Reply:
Right, our `moveCursor` does nothing, it just hands over to QTableView::moveCursor.
Ctrl+Left/Right allows moving the focus (table cell) while preserving the (multi-)selection.

With this branch I can move the focus around as before, and **only if I'm in the first column** Ctrl+left would move focus to the sidebar.
Is this how you intended this to work?

The other way around, Ctrl+Right moves focus from the sidebar to the tracks table, but there it restores the previous focus cell (expected, might be any column).
Therefore it's not possible to use Ctrl+Left/Right exactly as Tab / Shift+Tab.

Reply:
Ah, that is not how it is behaving on macOS. Here with ctrl (cmd on macOS) right / left switch between sidebar and library just like tab / shift+tab, independent of the column in which the focus is.

When moving with left/right **without** modifier key the full row stays selected anyway. Isn't that the case on Linux? If I select multiple rows (e.g. with shift up/down, or which ctrl-mouseclick), left/right **without** modifier key results in only the current row being selected. Shift left/right doesn't seem to do anything (except triggering a redraw of the waveforms for some reason)

Still, I don't really understand what "moving the focus (table cell) while preserving the (multi-)selection." implies. Is that referring to a multi-row selection? If so, I don't see the value of that at all. Do you? I have a hard time imagining a situation in which I would like to navigate the focus to a different cell while I am doing something with a multi row selection.

Reply:
> When moving with left/right without modifier key the full row stays selected anyway.

> I select multiple rows (e.g. with shift up/down, or which ctrl-mouseclick), left/right without modifier key results in only the current row being selected.

Same on Linux in both cases.

> till, I don't really understand what "moving the focus (table cell) while preserving the (multi-)selection." implies. Is that referring to a multi-row selection? If so, I don't see the value of that at all.

Yes, that's what I mean.
Well, the only thing coming to mind is using the currently focused column for sorting via CO (`[Library]sort_column_toggle` value `0`), though I can't tell if anyone would ever use it in such a scenario, probably not.

Reply:
IIUC the only way forward to enforce the intended focus back/forth behaviour would be block the native Ctrl+Left/Right cursor event and do the focus move, right?